pkg/scheduler/framework/plugins/interpodaffinity/filtering.go

```diff
@@ -229,35 +182,26 @@ func getTPMapMatchingExistingAntiAffinity(pod *v1.Pod, allNodes []*framework.Nod
 			return
 		}
 		for _, existingPod := range nodeInfo.PodsWithAffinity {
-			existingPodTopologyMaps, err := getMatchingAntiAffinityTopologyPairsOfPod(pod, existingPod.Pod, node)
-			if err != nil {
-				errCh.SendErrorWithCancel(err, cancel)
-				return
-			}
-			if existingPodTopologyMaps != nil {
+			existingPodTopologyMaps := getMatchingAntiAffinityTopologyPairsOfPod(pod, existingPod, node)
+			if len(existingPodTopologyMaps) != 0 {
 				appendResult(existingPodTopologyMaps)
 			}
 		}
 	}
-	parallelize.Until(ctx, len(allNodes), processNode)
-
-	if err := errCh.ReceiveError(); err != nil {
-		return nil, err
-	}
+	parallelize.Until(context.Background(), len(allNodes), processNode)
 
-	return topologyMap, nil
+	return topologyMap
 }
 
 // getTPMapMatchingIncomingAffinityAntiAffinity finds existing Pods that match affinity terms of the given "pod".
 // It returns a topologyToMatchedTermCount that are checked later by the affinity
 // predicate. With this topologyToMatchedTermCount available, the affinity predicate does not
 // need to check all the pods in the cluster.
-func getTPMapMatchingIncomingAffinityAntiAffinity(pod *v1.Pod, allNodes []*framework.NodeInfo) (topologyToMatchedTermCount, topologyToMatchedTermCount, error) {
+func getTPMapMatchingIncomingAffinityAntiAffinity(podInfo *framework.PodInfo, allNodes []*framework.NodeInfo) (topologyToMatchedTermCount, topologyToMatchedTermCount) {
 	topologyPairsAffinityPodsMap := make(topologyToMatchedTermCount)
 	topologyToMatchedExistingAntiAffinityTerms := make(topologyToMatchedTermCount)
-	affinity := pod.Spec.Affinity
-	if affinity == nil || (affinity.PodAffinity == nil && affinity.PodAntiAffinity == nil) {
-		return topologyPairsAffinityPodsMap, topologyToMatchedExistingAntiAffinityTerms, nil
+	if len(podInfo.RequiredAffinityTerms) == 0 && len(podInfo.RequiredAntiAffinityTerms) == 0 {
```

Comment:
would something fail if we just returned nil? just to avoid unnecessary allocations.

Reply:
there are more than few map access in the file, needs to be verified, I don't trust the unit tests to cover all cases.

Reply:
It'd safe to keep the map instantiated, as some places directly access it without check if it's nil.